vignettes/lock_and_prop.Rmd

```diff
@@ -0,0 +1,74 @@
+---
+title: "renv::Lock File Updates and Propagation"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{renv::Lock File Updates and Propagation}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This article will go over the following:
+
+* The process for requesting an update to the Lock Files, i.e. `renv.lock`, for the `{admiral}` family of packages.
+* How `{admiralci}` uses a propagation process to update `renv.lock` files across the family of `{admiral}` packages .
+
+## Updates to Lock Files 
+
+The intention of the `renv.lock` file is to create a harmonized development environment across our many different companies and collaborators.  The core team feels that if everyone is working with the same version of R and the same versions of R packages then communication around development issues will be much easier.  Unfortunately, this adoption of a `renv.lock` file also creates a few barriers for our development process.  First, if we **lock** in a version of R and R packages then we can not quickly adopt the latest and greatest of updates from new releases.  Second, how does a team decide to update a certain package or version of R and who gets to make that final call? Finally, when do we make updates to the lock file. Hopefully, the below will provide some guidance around these issues.  However, these process might change over time and admiral developers are encouraged to stay up to date with the latest processes and participate in the discussion.
+
+**Note:** Through GitHub Actions we continuously test admiral's integration with the latest 3 snapshots of dependent R packages on CRAN that are closest in date to latest 3 versions of R.  Please see [GitHub Actions/Workflows]( https://pharmaverse.github.io/admiraldev/devel/articles/pr_review_guidance.html#github-actionsworkflows) on the `{admiraldev}` site for more information on this process.
+
+### R Package Updates and Lock Files
+
+As you gathered, package updates to the `renv.lock` file are permissible and needed as our family grows and dependent packages and R versions change.  
+
+Most of the scenarios for updates can be slotted into the following:
+
+1. A new function in a dependent package, e.g. `{dplyr}` is available and could be utilized in a new admiral function.
+1. A new package is available with new functions that could greatly simplify our code base or add in additional functionality.
+1. A breaking bug is discovered in a dependent package that breaks backward compatibility for that package with admiral.
+1. Other?
```

Comment:
- a dependency is removed
- a new version of a dependent package causes R-CMD checks to fail, e.g., due to a deprecated function.